src/pretix/base/models/billing.py

```diff
@@ -30,6 +30,10 @@ class BillingInvoice(LoggedModel):
     currency = models.CharField(max_length=3)
 
     ticket_fee = models.DecimalField(max_digits=10, decimal_places=2)
+    final_ticket_fee = models.DecimalField(max_digits=10, decimal_places=2, default=0)
+    voucher_discount = models.DecimalField(max_digits=10, decimal_places=2, default=0)
+    voucher_price_mode = models.CharField(max_length=20, null=True, blank=True)
```

Comment:
Below, you compare this field to `InvoiceVoucher` enum. Why not limit the choices for this field?